src/shogun/base/SGObject.h

```diff
@@ -124,6 +124,12 @@ class CSGObject
 	/** destructor */
 	virtual ~CSGObject();
 
+	template<class Archive>
```

Comment:
Weeeeellll.
I mean so internally OK.
But for shogun users, can we please have functions that only take filenames and an enum that tells the file type? SWIG compatible etc.
I